### PR TITLE
Support loading openapi from []bytes

### PR DESCRIPTION
**Pull request by pwittrock**
Fixes: #2070

### CODE REVIEW

File: kyaml/openapi/openapi.go [hunk 1]
```diff
@@ -4,15 +4,23 @@
...
+	"sigs.k8s.io/kustomize/kyaml/errors"
 	"sigs.k8s.io/kustomize/kyaml/yaml"
 )
 
-var setup sync.Once
-var schema spec.Schema
-var schemaByResourceType map[yaml.TypeMeta]*spec.Schema
+var c state
+
+type state struct {
```
Comment (monopole) [inline kyaml/openapi/openapi.go]: some kind of struct comment please?  or at least a better name

File: kyaml/openapi/openapi.go [hunk 2]
```diff
@@ -26,13 +34,45 @@ type ResourceSchema struct {
...
+	rs, found := c.schemaByResourceType[t]
 	if !found {
 		return nil
 	}
 	return &ResourceSchema{Schema: rs}
 }
 
+// AddSchema parses s, and adds definitions from s to the global schema.
+func AddSchema(s []byte) error {
+	if c.schemaByResourceType == nil {
```
Comment (monopole) [inline kyaml/openapi/openapi.go]: i don't know why this `if` is here - it doesn't look at 's'
should this be in the ctor?

File: kyaml/openapi/openapi.go [hunk 3]
```diff
@@ -26,13 +34,45 @@ type ResourceSchema struct {
...
+		if err != nil {
+			return nil, errors.Wrap(err)
+		}
+		sc = *r
+	}
+
+	return &ResourceSchema{Schema: &sc}, nil
+}
+
+// NoUseBuiltInSchema can be called with b=false to prevent using the built-in Kubernetes
```
Comment (monopole) [inline kyaml/openapi/openapi.go]: passing bools is hard to understand at the call point

how about no setter, and just a noarg method like `suppressBuiltinSchemaUse()` (or the opposite)

File: kyaml/openapi/openapi.go [hunk 4]
```diff
@@ -190,11 +246,32 @@ func parse(b []byte) {
...
+	return nil
+}
+
+func resolve(root interface{}, ref *spec.Ref) (*spec.Schema, error) {
+	res, _, err := ref.GetPointer().Get(root)
+	if err != nil {
+		return nil, errors.Wrap(err)
+	}
+	switch sch := res.(type) {
+	case spec.Schema:
```
Comment (monopole) [inline kyaml/openapi/openapi.go]: i love Go switch statements.

File: kyaml/openapi/openapi.go (hunk 1 above)
Comment (pwittrock) [reply kyaml/openapi/openapi.go]: done

File: kyaml/openapi/openapi.go (hunk 2 above)
Comment (pwittrock) [reply kyaml/openapi/openapi.go]: moved into parse where this is used

File: kyaml/openapi/openapi.go (hunk 3 above)
Comment (pwittrock) [reply kyaml/openapi/openapi.go]: done

File: kyaml/openapi/openapi.go (hunk 4 above)
Comment (pwittrock) [reply kyaml/openapi/openapi.go]: yay

**Comment by pwittrock** [comment]
PTAL

**Comment by pwittrock** [comment]
Thanks!

**Pull request merged**